external-plugins/release-blocker/server.go

```diff
@@ -133,25 +149,103 @@ func (s *Server) handleEvent(eventType, eventGUID string, payload []byte) error
 	return nil
 }
 
-func (s *Server) handleLabel(label string, org string, repo string, num int, add bool) error {
-	hasLabel, err := hasLabel(s.GHC, label, org, repo, num)
+func branchExistsFunc(org string, repo string, targetBranch string) (bool, error) {
+
+	resp, err := http.Head(fmt.Sprintf("https://github.com/%s/%s/tree/%s", org, repo, targetBranch))
 	if err != nil {
-		return err
+		return false, err
+	}
+
+	if resp.StatusCode == 200 {
+		return true, nil
+	} else if resp.StatusCode == 404 {
+		return false, nil
+	}
+
+	return false, fmt.Errorf("Failed to detect branch %s, got http response code %d", targetBranch, resp.StatusCode)
+}
+
+func (s *Server) handleLabel(targetBranch string, org string, repo string, num int, add bool) (string, error) {
+
+	label := fmt.Sprintf("%s/%s", baseLabel, targetBranch)
+
+	hasLabel, err := hasLabel(s.ghc, label, org, repo, num)
+	if err != nil {
+		return "", err
 	}
 
 	if !hasLabel && add {
-		err := s.GHC.AddLabel(org, repo, num, label)
+		exists, err := s.branchExists(org, repo, targetBranch)
 		if err != nil {
-			return err
+			return "", err
+		}
+
+		if !exists {
+			return fmt.Sprintf("Unable to place blocker label for release branch [%s] because branch does not exist.", targetBranch), nil
+		}
+
+		err = s.ghc.AddLabel(org, repo, num, label)
+		if err != nil {
+			return "", err
 		}
 	} else if hasLabel && !add {
-		err := s.GHC.RemoveLabel(org, repo, num, label)
+		err := s.ghc.RemoveLabel(org, repo, num, label)
+		if err != nil {
+			return "", err
+		}
+	}
+
+	return "", nil
```

Comment:
Just being curious: are you returning an empty string here to avoid any errors regarding `nil`. Is that also considered good style in Go as it is in other languages?

Reply:
I wanted to separate API errors (which are returned in the second argument) from github comment responses (which are returned in the first arg). Those are being treated independently from one another